client/model/pillarModel.js

```diff
@@ -185,10 +196,18 @@ define( [
                     if (typeof pair.value !== 'undefined' && pair.value !== "") {
                         try {
                             // make sure we can parse it later
-                            JSON.parse(pair.value);
+                            jsonlint.parse(pair.value);
                         } catch(err) {
-                            swal("Error", "Please make sure your values consist of valid JSON", 'error');
+                            if (err.name == 'TypeError') {
+                                err.message = "No value provided for key " + pair.key + ". Remove key-value pair if not " +
+                                "needed.";
+                            }
                             ret = false;
+                            // save the first error we saw so end user can tell which line is invalid
+                            if (first_error == "") {
```

Comment:
the idea behind this was to expose only one error to the user at a time, top-down. so if the first row has an error and so does the second, the first row error will be shown until fixed - and then the second will be shown.